### PR TITLE
Feature/v3.17.x.x/ high dpi support

### PR DESCRIPTION
**Pull request by Szczyrk**

### CODE REVIEW

**Review by genail — APPROVED** [review]
Looks good!

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 1]
```diff
@@ -117,10 +122,38 @@ private Vector2 GetScreenResoultion()
         private void Awake()
         {
             _logger = PatcherLogManager.DefaultLogger;
+            _canvasScaler = GetComponent<CanvasScaler>();
+
+            float scaler;
+            float currentDpi = Screen.dpi;
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs:128]: I'd recommend renaming it to `float screenDpi`.

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 2]
```diff
@@ -117,10 +122,38 @@ private Vector2 GetScreenResoultion()
         private void Awake()
         {
             _logger = PatcherLogManager.DefaultLogger;
+            _canvasScaler = GetComponent<CanvasScaler>();
+
+            float scaler;
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs:127]: I'd recommend renaming it to `float screenScale`.

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 3]
```diff
@@ -117,10 +122,38 @@ private Vector2 GetScreenResoultion()
...
+                scaler = 2;
+            }
+            else if (currentDpi >= 400)
+            {
+                scaler = 4;
+            }
+            else
+            {
+                scaler = 1;
+            }
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs:145]: I'd recommend different flow of condition checking

```
if (currentDpi >= 400)
{
    ...
}
else if (currentDpi >= 200)
{
    ...
}
else
{
    if (currentDpi == 0)
    {
        ... (log warning)
    }

    ...
}
```

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 4]
```diff
@@ -142,7 +175,7 @@ private void Awake()
 #endif
         }
 
-        private void EnforceCorrectScreenSize()
+        private void EnforceCorrectScreenSize(float scaler)
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs:178]: I'd recommend renaming argument to `float screenScale`.

**Review by witcher112 — CHANGES_REQUESTED** [review]

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs (hunk 1 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: done

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs (hunk 2 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: done

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs (hunk 3 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: done

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 5]
```diff
@@ -117,10 +122,38 @@ private Vector2 GetScreenResoultion()
...
+            {
+                scaler = 4;
+            }
+            else
+            {
+                scaler = 1;
+            }
 
-            EnforceCorrectScreenSize();
+            _logger.LogDebug(string.Format("Scaler resolution: {0}", scaler));
```
[the remark this answers is not in the document]
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: done

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs (hunk 4 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: done

File: Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs [hunk 6]
```diff
@@ -19,6 +22,8 @@ public class BorderlessWindow : MonoBehaviour
 
         public Rect DraggableArea;
 
+        private CanvasScaler _canvasScaler;
```
[the remark this answers is not in the document]
Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: Not really, the second option requires having both `BorderlessWindow` and `CanvasScaler` on the same game object, while the public field + `FindObjectOfType` allows having them separated.

[the remark this answers is not in the document]
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/UI/BorderlessWindow.cs]: ok done

**Review by witcher112 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## [3.17.7.0]
 ### Added
 - Checking path length on Windows
+- Support high dpi
```
[the remark this answers is not in the document]
Comment (Szczyrk) [reply CHANGELOG.md]: done

**Pull request merged**